duckduckgo/Android
# Feature/tracker blocking opt in

Task/Issue URL: https://app.asana.com/0/inbox/608920331025313/1121943454981948/1123477418692919
Tech Design URL: 
CC: 

**Description**:
Adds a tracker-blocking opt-in screen as the first page of the onboarding flow.

**Steps to test this PR**:
1. As an existing user with tracker blocking **enabled**, install this version
    1. confirm you **don't see** onboarding screens
    1. confirm you still have tracker blocking enabled (from the privacy dashboard)
1. As an existing user with tracker blocking **disabled**, install this version
    1. confirm you **don't see** onboarding screens
    1. confirm you don't have tracker blocking enabled (from the privacy dashboard)

1. Perform a fresh install, and ensure you receive variant **mm** (control group)
    1. Launch the app
    1. Ensure you **do not** see the tracker blocking opt in screen
    1. Verify that tracker blocking is enabled (in privacy dashboard)

1. Perform a fresh install, and ensure you receive variant **mn** (show tracker blocking opt in page)
    1. Launch the app
    1. Ensure you **do see** the tracker blocking opt in screen
    1. Choose to activate tracker blocking
    1. Verify that the subtitle reads as "Stay private as you search and browse the web" 
    1. Verify shield icon is blue
    1. Verify that "Browsing Protection" subtitle is "Block trackers and browse securely by default"
    1. Verify that tracker blocking is enabled (in privacy dashboard)

1. Perform a fresh install, and ensure you receive variant **mn** (show tracker blocking opt in page)
    1. Launch the app
    1. Ensure you **do see** the tracker blocking opt in screen
    1. Choose to **not activate** tracker blocking
    1. Verify that you see an alert-style dialog, with red shield, encouraging user to return
    1. Choose to "continue" without protection
    1. Verify that the subtitle reads as "For full browsing protection, go back to activate tracker blocking" 
    1. Verify shield icon is red
    1. Verify that "Browsing Protection" subtitle is "Without tracker blocking you don't have full browsing protection."
    1. Continue through onboarding
    1. Verify that tracker blocking is disable (in privacy dashboard)

Additionally:
    
1. Check back button behaviour
    1. Back button should navigate _back_ to previous page
    1. Back button should dismiss the opt-out conformation dialog (as should tapping outside of it, 

add hitting the on-screen back buttons

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
